### ISSUE TITLE
Context references fail for BridgeDomain

### ISSUE THREAD

**Issue by daniel-zuckerberg** [datacenter/acitoolkit#41]
All calls fail to get or set the Context in the BridgeDomain class for acitoolkit.py. All get calls return "None".

**Comment by michsmit99** [comment]
Hi Daniel, can you post what you are trying to do ?  It seems to work for me.  The add_context and get_context only work on the local objects.  If you want to create config on the APIC, the changes will remain local until you call push_to_apic.  Likewise to get the current objects from the APIC, first call Tenant.get_deep(session, <tenant_name>) to load the objects locally and then traverse using get_context, get_children, etc.

**Comment by daniel-zuckerberg** [comment]
Michael, I must be doing something wrong then. I'm trying assign a context to a Bridge Domain with the CLI app. Here's what I'm executing:

fabric#
fabric# conf t
fabric(config)# tenant PRODUCTION
Executing create tenant command
fabric(config)# exit
fabric# switchto PRODUCTION
fabric-PRODUCTION# conf t
fabric-PRODUCTION(config)# context AMERICAS
Executing create context command
fabric-PRODUCTION(config-ctx)# exit
fabric-PRODUCTION(config)# bridgedomain VLAN_207
Executing create bridgedomain command
fabric-PRODUCTION(config-bd)# ip address 10.10.207.254/24
Executing create subnet command
fabric-PRODUCTION(config-bd)# context AMERICAS

My context never gets assigned to the Bridge Domain. Looking though the code, I don’t see this built out yet in the BridgeDomainConfigSubMode class in acitoolkitcli.py. So I added the following to do_context() to try to get this working:

def do_context(self, args):
     self.bridgedomain.add_context(args[0])
     resp = self.apic.push_to_apic(self.tenant.get_url(), self.tenant.get_json())
     if not resp.ok:
          error_message(resp)
     return

But still no luck. And if I go and manually set the context, I notice that "self.bridgedomain.get_context()" is returning “None”.  Even though it is assigned.

**Comment by michsmit99** [comment]
Ah, makes sense.  I was assuming you were writing a script.  Yes, the context command implementation is missing in the CLI app.  The reason your code isn't working is that the add_context call expects to receive a Context object not a string with the context name.  I checked in a fix based on your code.  I also added some assertions within the add_context method to check that the context is indeed an instance of the Context class.
